### ISSUE TITLE
Option to preview needed items for schematics in your inventory, before building

### ISSUE THREAD

**Issue by dominikaaaa** [cabaletta/baritone#1781]
## Describe your suggestion
When you build and the build fails because you don't have all the blocks, adding a ghost-like semi-transparent texture of the needed items in your inventory would be nice. 

## Settings
`#previewBuildItems`

## Context
Most users are confused when building fails, adding a visual way for them to understand how would be immensely helpful in reducing the amount of stupid issues that are created. 

## Final checklist
- [x] I know how to properly use check boxes 
- [x] I have not used any OwO's or UwU's in this issue.

**Comment by UltraBlackLinux** [comment]
Maybe just a list of items in chat, if you click on one item it mines it

**Comment by Dryhb** [comment]
There is an existing mod that can make this type of "material list" that appears as a hud on the screen and display what items you need to acquire to have all the mats. It can handle shulkerboxes with items in it. The name is Litematica (similar version of Schematica but more developed and much better).
It's a mod by masady, you can find it on github too or in the scicraft discord for more infos

**Comment by dominikaaaa** [comment]
Litematica isn't compatible with Baritone to my knowledge, and doesn't run on Forge.

**Comment by Dryhb** [comment]
yes it's incompatible with Baritone but runs on forge 1.12.2 but I just wanted to point out that it's possible and maybe the code is public

**Comment by EntityinArray** [comment]
Yes, litematica+baritone would be an awesome auto-building combo!

**Comment by tr432** [comment]
You can use [Schematic2Blueprint](https://double-helix.industries/applications/schematic2blueprint/) in order to do this.
